streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java

```diff
@@ -192,6 +192,10 @@ public boolean isValidTransition(final State newState) {
 
     void markChangelogAsCorrupted(final Collection<TopicPartition> partitions);
 
+    void recordProcessLatency(final long elapsedLatencyMs);
```

Comment:
This is a bit awkward.. if anyone have a more elegant solution I'm all ears.

Reply:
prop: You could record the time spent in processing in `Task#process()` and just have a method `Task#recordProcessRatio(final long totalProcessTime)` in the interface.

Reply:
Hey @guozhangwang , I did have some ideas (similar to @cadonna 's), but it might be easier to just present it as a diff: https://github.com/guozhangwang/kafka/pull/12

Reply:
I originally did this, but I'm concerned that the latency for processing a single record is sub-milliseconds and hence the cumulated latency would be zero.